src/reporter.js

```diff
@@ -220,11 +220,14 @@ function CypressXML(runner, options) {
         relative = relative.substring(0, relative.lastIndexOf('.')) + config.logFileExt;
         var logFile = path.join(config.logsFolder, relative);
         if (fs.existsSync(logFile)) {
-          logContent = fs.readFileSync(logFile);
+          let fileContent = fs.readFileSync(logFile);
+          logContent = fileContent.toString();
         }
       }
       var videoFile = path.join(config.videosFolder, specRelativePath)+'.mp4';
-      logContent += '[[ATTACHMENT|' + videoFile +']]';
+      if (suiteStats.failures > 0) {
+        logContent += '[[ATTACHMENT|' + videoFile +']]';
+      }
```

Comment:
The `if` condition should better be before the `var videoFile`.

Also, it would be good to add `\n` before the `[[ATTACH...`

Reply:
I guess, you didn't notice my second sentence?